frontend/src/components/filtertable.js

```diff
@@ -160,3 +169,135 @@ export class FilterTable extends React.Component {
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      fieldSelection: null,
+      isFieldOpen: false,
+      isValueOpen: false,
+      valueOptions: [],
+      valueSelections: [],
+    };
+  }
+
+  onFieldToggle = isExpanded => {
+    this.setState({isFieldOpen: isExpanded})
+  };
+
+  onValueToggle = isExpanded => {
+    this.setState({isValueOpen: isExpanded})
+  };
+
+  onFieldSelect = (event, selection) => {
+    this.setState(
+      // clear value state too, otherwise the old selection remains selected but is no longer visible
+      {fieldSelection: selection, isFieldOpen: false, valueSelections: [], valueOptions: [], isValueOpen: false},
+      this.updateValueOptions
+    )
+
+  };
+
+  onValueSelect = (event, selection) => {
+    // update state and call setFilter
+    const valueSelections = this.state.valueSelections;
+    let updated_values = (valueSelections.includes(selection))
+      ? valueSelections.filter(item => item !== selection)
+      : [...valueSelections, selection]
+
+    this.setState(
+      {valueSelections: updated_values},
+      () => this.props.setFilter(this.state.fieldSelection, this.state.valueSelections.join(';'))
+    )
+  };
+
+  onFieldClear = () => {
+    this.setState(
+      {fieldSelection: null, valueSelections: [], isFieldOpen: false, isValueOpen: false},
+    )
+  };
+
+  onValueClear = () => {
+    this.setState(
+      {valueSelections: [], isValueOpen: false},
+      () => this.props.setFilter(this.state.fieldSelection, this.state.valueSelections)
+    )
+  }
+
+  updateValueOptions = () => {
+    const {fieldSelection} = this.state
+    const {customFilters} = this.props
+
+    console.log('CUSTOMFILTER: '+customFilters)
+    if (fieldSelection !== null) {
+      let api_filter = toAPIFilter(customFilters).join()
+      console.log('APIFILTER: '+customFilters)
+
+      fetch(buildUrl(Settings.serverUrl + '/widget/result-aggregator',
+        {
+          group_field: fieldSelection,
+          run_id: this.props.runId,
+          additional_filters: api_filter,
+        }
+      ))
+      .then(response => response.json())
+      .then(data => {
+        this.setState({valueOptions: data})
+      })
+    }
+  }
+
+  render () {
+    const {isFieldOpen, fieldSelection, isValueOpen, valueOptions, valueSelections} = this.state
+    let field_selected = this.state.fieldSelection !== null
+    let values_available = valueOptions.length > 0
+    let value_placeholder = "Select a field first"  // default instead of an else block
+    if (field_selected && values_available){ value_placeholder = "Select value(s)"}
+    else if (field_selected && !values_available) { value_placeholder = "No Values for selected field"}
+    return (
+      <React.Fragment>
+        <Select key="metafield_select"
+          aria-label="metadata-field-filter"
+          placeholderText="Select Metadata Field"
```

Comment:
To match casing of the other select:
```suggestion
          placeholderText="Select metadata field"
```